glib/src/subclass/types.rs

```diff
@@ -876,20 +876,54 @@ pub(crate) unsafe fn signal_override_class_handler<F>(
 ) where
     F: Fn(&super::SignalClassHandlerToken, &[Value]) -> Option<Value> + Send + Sync + 'static,
 {
+    let (signal_id, _) = SignalId::parse_name(name, from_glib(type_), false)
+        .unwrap_or_else(|| panic!("Signal '{}' not found", name));
+
+    let query = signal_id.query();
+    let return_type = query.return_type();
+
     let class_handler = Closure::new(move |values| {
         let instance = gobject_ffi::g_value_get_object(values[0].to_glib_none().0);
-        class_handler(&super::SignalClassHandlerToken(instance as *mut _), values)
+        let res = class_handler(
+            &super::SignalClassHandlerToken(
+                instance as *mut _,
+                return_type.into(),
+                values.as_ptr(),
+            ),
+            values,
+        );
+
+        if return_type == Type::UNIT {
```

Comment:
Considering it's the exact same code as above, do you think it's worth it to write a function where we make this check instead of duplicating the code?

Reply:
As it's just in two places, and those are in different modules, having a function shared between both modules for that seems a bit overkill

Reply:
Fine by me!